calculator.rb

```diff
@@ -0,0 +1,106 @@
+# 2/4/2020
+# Ada Classwork by Nataliya Pogodina
+
+def print_operation_instructions
+  puts "Welcome to Awesome Ruby Calculator"
+  puts "=================================="
+  puts "Please, choose a math operation:"
+  puts "1. add(+)"
+  puts "2. substract(-)"
+  puts "3. multiply(*)"
+  puts "4. divide(/)"
+  puts "(type a name or a symbol)"
+end
+
+def get_math_operation
+  operation_types = ["add", "+", "substract", "-", "multiply", "*", "divide", "/"]
+  math_operation = gets.chomp.downcase
+
+  until operation_types.include?(math_operation)
+    puts "Please, enter a valid operation!"
+    puts "(review the guide above!)"
+    math_operation = gets.chomp
+  end
+  return math_operation
+end
+
+
+def validate(input)
+  while input !~ /\A[+-]?\d+(\.\d+)?\z/ # regex
+    print "Please, enter a valid number: "
+    input = gets.chomp
+  end
+  return input.to_f
+end
+
+# Understanding the regex
+# /         start of the regex
+# \A        start of the string to be matched
+# [+-]?     zero or one of '+' or'-'
+# \d+       one or more of digit
+# (\.\d+)?  zero or one of 'one dot and 'one or more of digit'' 
+# \z        end of the string to be matched
+# /         end of the regex
+
+# def check_if_int(input)
```

Comment:
Minor note: you should remove unused code from your finished project. Consider creating a file for yourself to store code logic that you'd like to save and might use in the future.